addonscript/src/addonscript/api.py

```diff
@@ -14,6 +14,36 @@
 UPLOAD_STATUS_PK = "api/v3/addons/{id}/versions/{version}/uploads/{upload_pk}/"
 
 
+# https://addons-server.readthedocs.io/en/latest/topics/api/applications.html
+# XXX when we want to support multiple products, we'll need to unhardcode the
+#     `firefox` below
+ADD_VERSION = "api/v4/applications/firefox/{version}/"
+
+
+async def add_version(context, version):
+    """Add a new version to AMO.
+
+    Use the `min_version` here, rather than the string with the buildid etc.
+
+    Raises:
+        BadVersionError: If the XPI's version fails to sanity check against AMO
+
+
+    """
+    url = get_api_url(context, ADD_VERSION, version=version)
+
+    try:
+        result = await amo_put(context, url, data=None)
+    except ClientResponseError as exc:
+        if exc.status == 401:
+            raise AuthFailedError("Addonscript credentials are misconfigured")
+        elif exc.status == 403:
+            raise AuthInsufficientPermissionsError("Addonscript creds are missing permissions")
+        raise exc
```

Comment:
In the future, we could log if the version was created or already existed by the 20x response. As you mentioned, this would require different behavior from `amo_put` or a separate function.

Reply:
👍